velox/common/base/AbstractBloomFilter.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+class AbstractBloomFilter {
+ public:
+  virtual bool mightContain(int32_t value) const = 0;
```

Comment:
does that mean that checking existence will trigger a virtual call? Won't this be terribly slow?

Reply:
This is done once per row group, so it should be ok? Also, we are creating method 
```
virtual bool testBloomFilter(
      const AbstractBloomFilter& bloomFilter,
      const velox::Type& type) 
```
in the `Filter` class so, we need to have an abstract type defined.